nixos/modules/programs/steam.nix

```diff
@@ -4,13 +4,31 @@ with lib;
 
 let
   cfg = config.programs.steam;
+  gsCfg = config.programs.gamescope;
 
-  steam = pkgs.steam.override {
+  steam = pkgs.steam.override ({
     extraLibraries = pkgs: with config.hardware.opengl;
       if pkgs.hostPlatform.is64bit
       then [ package ] ++ extraPackages
       else [ package32 ] ++ extraPackages32;
-  };
+  } // optionalAttrs
+    (cfg.gamescopeSession && gsCfg.capSysNice) {
+      buildFHSUserEnv = pkgs.buildFHSUserEnvBubblewrap.override {
+        # use the setuid wrapped bubblewrap
+        bubblewrap = "/run/wrappers";
+      };
+    }
+  );
+
+  gamescopeSessionFile = (pkgs.writeTextDir "share/wayland-sessions/steam.desktop" ''
+    [Desktop Entry]
+    Name=Steam
+    Comment=A digital distribution platform
+    Exec=${pkgs.writeShellScript "steam-gamescope" ''
+      gamescope -- steam -tenfoot -pipewire-dmabuf
+    ''}
+    Type=Application
```

Comment:
Please use makeDesktopItem

Reply:
I'm not sure if that's correct in this case, since the desktop file must live at "share/wayland-sessions/steam.desktop", and `makeDesktopItem` hardcodes ["share/applications"](https://github.com/NixOS/nixpkgs/blob/master/pkgs/build-support/make-desktopitem/default.nix#L115) which is useless for `services.xserver.displayManager.sessionPackages`.

Although, perhaps it would be useful to modify `makeDesktopItem` to allow for overriding the destination?

Reply:
> services.xserver.displayManager.sessionPackages

Do you know of any other usecases that need a different place for the desktop file? If there are multiple we should definitely add an option to change the directory otherwise I am a bit undecided.

Reply:
Just this one option that I'm aware of but, there are two possible locations "share/wayland-sessions" and "share/xsessions"

Reply:
Having the option to override the location of the produced desktop file definitely seems useful to me, even if nothing else in nixpkgs uses it now.